pallets/portfolio/src/lib.rs

```diff
@@ -329,4 +353,175 @@ impl<T: Trait> Module<T> {
         ensure!(name_uniq, Error::<T>::PortfolioNameAlreadyInUse);
         Ok(())
     }
+
+    /// Accept custody of a Portfolio
+    fn accept_portfolio_custody(new_custodian: IdentityId, auth_id: u64) -> DispatchResult {
+        ensure!(
+            <identity::Authorizations<T>>::contains_key(Signatory::from(new_custodian), auth_id),
+            AuthorizationError::Invalid
+        );
+
+        let auth = <identity::Authorizations<T>>::get(Signatory::from(new_custodian), auth_id);
+
+        let (portfolio_owner, portfolio_number) = match auth.authorization_data {
+            AuthorizationData::PortfolioCustody(did, number) => (did, number),
+            _ => return Err(Error::<T>::IrrelevantAuthorization.into()),
+        };
+
+        let portfolio_id = Self::get_portfolio_id(portfolio_owner, portfolio_number);
+
+        if let Some(num) = portfolio_number {
+            ensure!(
+                <Portfolios>::contains_key(portfolio_owner, num),
+                Error::<T>::PortfolioDoesNotExist
+            );
+        }
+
+        let current_custodian = if <PortfolioCustodian>::contains_key(&portfolio_id) {
+            <PortfolioCustodian>::get(&portfolio_id)
+        } else {
+            portfolio_owner
+        };
+
+        <identity::Module<T>>::consume_auth(
+            current_custodian,
+            Signatory::from(new_custodian),
+            auth_id,
+        )?;
+
+        <PortfolioCustodian>::insert(&portfolio_id, new_custodian);
+
+        Self::deposit_event(RawEvent::PortfolioCustodianChanged(
+            new_custodian,
+            portfolio_id,
+            new_custodian,
+        ));
+        Ok(())
+    }
+
+    pub fn unchecked_transfer_portfolio_balance(
```

Comment:
Maybe worth noting here that if there would have been an underflow but no overflow or vice versa that a token balance would have been created or destroyed out of thin air.

Reply:
Makes sense. FWIW it is named `unchecked` to signify that it does not do data validation checks and anything that causes an underflow/overflow should not have been sent here. I still did saturating add/sub to be defensive.